lib/saluki-components/src/destinations/datadog/status_flare/mod.rs

```diff
@@ -0,0 +1,195 @@
+            .replace("_", "-")
+            .to_lowercase();
+        let api_listen_port = config
+            .try_get_typed::<NonZeroUsize>("remote_agent_api_listen_port")?
+            .unwrap_or(DEFAULT_API_LISTEN_PORT);
+        let client = RemoteAgentClient::from_configuration(config).await?;
+
+        Ok(Self {
+            id: format!("{}-{}", formatted_full_name, Uuid::now_v7()),
+            display_name: formatted_full_name,
+            api_listen_port,
+            client,
+        })
+    }
+}
+
+#[async_trait]
+impl DestinationBuilder for DatadogStatusFlareConfiguration {
+    fn input_data_type(&self) -> DataType {
+        DataType::Metric | DataType::EventD | DataType::ServiceCheck
+    }
+
+    async fn build(&self, _context: ComponentContext) -> Result<Box<dyn Destination + Send>, GenericError> {
+        Ok(Box::new(DatadogStatusFlare {
+            id: self.id.clone(),
+            display_name: self.display_name.clone(),
+            api_listen_port: self.api_listen_port,
+            client: self.client.clone(),
+        }))
+    }
+}
+
+impl MemoryBounds for DatadogStatusFlareConfiguration {
+    fn specify_bounds(&self, builder: &mut MemoryBoundsBuilder) {
+        builder
+            .minimum()
+            // Capture the size of the heap allocation when the component is built.
+            .with_single_value::<DatadogStatusFlare>();
+    }
+}
+
+pub struct DatadogStatusFlare {
+    id: String,
+
+    display_name: String,
+
+    api_listen_port: NonZeroUsize,
+
+    client: RemoteAgentClient,
+}
+
+#[async_trait]
+impl Destination for DatadogStatusFlare {
+    async fn run(mut self: Box<Self>, mut context: DestinationContext) -> Result<(), GenericError> {
+        let Self {
+            id,
+            display_name,
+            api_listen_port,
+            mut client,
+        } = *self;
+
+        let api_endpoint = format!("127.0.0.1:{}", api_listen_port);
+        let auth_token: String = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(64)
+            .map(char::from)
+            .collect();
+
+        let mut register_agent = interval(Duration::from_secs(10));
+        register_agent.set_missed_tick_behavior(MissedTickBehavior::Delay);
+
+        let mut health = context.take_health_handle();
+        health.mark_ready();
+        debug!("Datadog Status and Flare destination started.");
+
+        loop {
+            select! {
+                _ = health.live() => continue,
+
+                result = context.events().next() => match result {
+                    Some(_events) => {
+                    },
+                    None => break,
+                },
+
+                // Time to (re)register with the Core Agent.
+                //
+                // TODO: Consider spawning the registration as a task so that the component can keep polling and not slow down the accepting of events and responding of health checks.
+                _ = register_agent.tick() => {
```

Comment:
This component is only built if `adp_standalone_mode == false`, but this call will keep failing if 

```
 remote_agent_registry:
   enabled: true
```
is not set on the `datadog-agent` side.

Reply:
🤷🏻 

I think that's just something we'll have to deal with until we make it the default to enable the remote agent registry. We don't get any visible errors, right, since it's just debug logging if the call fails? Or does it have any other user-visible impact in ADP when the remote agent registry is disabled?

Reply:
Correct, it will just spam the debug logs if it fails.